assets/prototype/domain/eventEditor/containers/queries/useCacheRehydration.js

```diff
@@ -0,0 +1,84 @@
+import { useEffect } from '@wordpress/element';
+import { useApolloClient } from '@apollo/react-hooks';
+import useCacheRehydrationData from './useCacheRehydrationData';
+import useRelations from '../../../../infrastructure/services/relations/useRelations';
+import useStatus from '../../../../infrastructure/services/status/useStatus';
+import useEventId from './useEventId';
+import { queries } from './';
+
+const { GET_TICKETS, GET_DATETIMES, GET_PRICE_TYPES, GET_PRICES } = queries;
+
+const useCacheRehydration = () => {
+	const client = useApolloClient();
+	const eventId = useEventId();
+	const { setData } = useRelations();
+	const { datetimes, tickets, prices, priceTypes, relations } = useCacheRehydrationData();
+	const { setIsLoaded } = useStatus();
+
+	useEffect(() => {
+		let { nodes = [] } = priceTypes;
+		if (nodes.length) {
+			client.writeQuery({
+				query: GET_PRICE_TYPES,
+				data: {
+					priceTypes,
+				},
+			});
+		}
+		setIsLoaded('priceTypes', true);
+
+		({ nodes = [] } = datetimes);
+		if (nodes.length) {
+			client.writeQuery({
+				query: GET_DATETIMES,
+				variables: {
+					where: {
+						eventId,
+					},
+				},
+				data: {
+					datetimes,
+				},
+			});
+		}
+		setIsLoaded('datetimes', true);
+
+		const datetimeIn = nodes.map(({ id }) => id);
+		({ nodes = [] } = tickets);
```

Comment:
not sure what it does

Reply:
destructuring by using an existing variable i.e. without `let` or `const` when destructuring. You need to wrap it in `()` to make it an expression.

Reply:
too much magic involved, can this be accomplished in other ways?

Reply:
from what you describe it sounds like the logic might be split into 2 or more small functions

Reply:
```js
let { name } = person;
```
is same as
```js
let name;
({ name} = person);
```
That way you can reuse an already declared variable